docs/DeveloperGuide.md

```diff
@@ -487,6 +487,84 @@ The following activity diagram summarizes what happens in the `MainWindow` class
+When a `ViewScheduleCommand` is executed with a given `keyDate`, where `keyDate` refers to the date on which the user wants to view schedule, `addPossibleGhostTasksWithMatchingDate(keyDate)` is responsible for this addition of relevant GhostTasks.
+
+Each task in the task list goes through a series of checks and actions before a GhostTask is created.
+Given below is an activity diagram that summarizes the sequence of  checks and actions taken for each task in the task list upon calling the above-mentioned method.
+
+![AddPossibleGhostTasksWithMatchingDateActivityDiagram](./images/AddPossibleGhostTasksWithMatchingDateActivityDiagram.png)
+
+Since the remaining general mechanisms by which the view schedule operation occurs, such as how the command is parsed and how the `ViewScheduleCommand` is created,  is similar to other previously elaborated commands, a step-by-step elaboration is not given for the overall execution.
+
+<br>
+
+#### Design Considerations
+**Aspect: Differentiating `RealTask` and `GhostTask` Objects:**
+* **Alternative 1:** Add a new field to `Task` objects that determine whether a task is a real task or not.
+    * Pros: Easier to implement and integrate with existing AB3 code
+    * Cons: Increased failure points, as an additional field has to be stored in the hard disk to determine if tasks are real or not. This field has to be kept track of in between different commands, but not exposed to the user.
+* **Alternative 2:** Make `Task` abstract and add concrete `RealTask` and `GhostTask` subclasses to it.
+    * Pros: Clearer classification of Task types. Polymorphism can be used to handle `RealTask` and `GhostTask` objects respectively.
+    * Cons: All code for existing `Task` objects needs to be refactored. More code needs to be written, which could result in more room for bugs.
+
+**Decision:**
+Alternative 2 was chosen as although Alternative 1 is simpler to implement, Alternative 1 has poor encapsulation of real and temporary task objects. `GhostTasks` need to be handled differently
+from `RealTasks`, as we do not want to expose them to the user. Hence, it makes more sense to encapsulate it as a separate class, even though more code needs to be refactored, written and tested.
+This also keeps the data stored in the hard disk smaller, as there is no unnecessary field to keep track of whether a task is real or not.
+
+<br>
+
+**Aspect: Searching of future occurrences of recurring tasks:**
+With recurring tasks, they can imply the existence of infinite potential future occurrences. Consequently, users could input dates well beyond reasonable amounts, such as centuries into
+the future. However, it is not sensible nor feasible to search for such extraneous lengths of time. Hence, the maximum number of time that a user can view schedule on a future date has been
+capped to 12 weeks, or 84 days, from the current date.
```

Comment:
Perhaps 'maximum number of time' can be rephrased? And I think it should be 'capped at' instead of 'capped to'?

Reply:
Ah this was meant to be 'maximum amount of time'. Will update it!